src/blog/2023/06/node-red-ha-flowforge.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Exploring High Availability for Node-RED in FlowForge
+subtitle: Exploring the Node-RED High Availability preview feature included in FlowForge 1.8.
+description: Exploring the Node-RED High Availability preview feature included in FlowForge 1.8.
+date: 2023-06-09 16:00:00
+authors: ["rob-marcer"]
+image: /images/blog/ha-test-header.jpg
+tags:
+    - posts 
+    - node-red
+---
+
+As part of FlowForge's 1.8 release, we have delivered our first step towards Highly Available (HA) Node-RED applications. In this article I'm going build an application to test the performance of Node-RED running in HA and standard mode.
+<!--more-->
+
+### Setting The Scene
+
+HA is our first [preview feature](https://flowforge.com/handbook/product/versioning/#preview-features), and your feedback is crucial to the future development path we will take. We encourage you to try out HA in your Node-RED instances and share your experiences with us.
+
+Our CTO Nick O'Leary, [wrote an article](/blog/2023/05/bringing-high-availability-to-node-red/) earlier this year exploring some of the terminology, concepts, and challenges of delivering HA for Node-RED. It's worth a read if you are new to HA or would like a refresher.
+
+This article is not an attempt to fully load-test FlowForge's performance with HA, the loads used were synthetic but should correlate strongly with real world performance of Node-RED applications.
+
+### How does the preview build of FlowForge's HA work?
+
+When HA is enabled in FlowForge for a Node-RED instance, various things change under the hood. We need to consider these changes when writing Node-RED flows for HA. The main points we need to consider are:
+
+##### HA runs multiple copies of the same instance
+
+Once HA is enabled, FlowForge creates multiple copies of your instance (2 at the time of writing). These copies all run in parallel. Any traffic to your instance will be load-balanced so that the traffic is shared between all copies.
+
+The load-balanced approach to HA creates a challenge when designing your applications, because it means you cannot:
+
+* determine which copy of your flows is dealing with a given message.
+* configure the copies of the flows to communicate to each other directly.
+* determine how many copies are running at any given moment in time.
```

Comment:
I don't understand this last point? What do you mean by 'copies are running' and why is this important?